restli-server/src/test/java/com/linkedin/restli/internal/server/response/TestBatchFinderResponseBuilder.java

```diff
@@ -341,4 +344,62 @@ private static Foo generateMetaData(Boolean onError)
     Foo foo = new Foo(map);
     return foo;
   }
+
+  @DataProvider(name = "build_uri_test_cases")
+  public Object[][] testBuildCriteriaURIDataProvider() throws RestLiSyntaxException
+  {
+    String[] requestUrls =
+    {
+      "/greetings?bq=searchGreetings&criteria=List((id:1,tone:SINCERE))&message=hello",
+      "/talent/api/talentHiringProjectCandidates?bq=candidates&candidates=List((candidate:urn%3Ali%3Ats_hire_identity%3A88156577))",
+      "/talent/api/talentHiringProjectCandidates?bq=candidates&candidates=List((candidate:urn%3Ali%3Ats_hire_identity%3A88156577))&fields=candidate",
```

Comment:
These tested URL has not:
1. Be real batch criterias at all (just single criteria) I will add to later test
2. These URLs are not exhaustive, I will add more URL test cases